pulp_ansible/app/tasks/collections.py

```diff
@@ -119,21 +119,20 @@ def import_collection(
     """
     CollectionImport.objects.get_or_create(task_id=get_current_job().id)
 
-    artifact = Artifact.objects.get(pk=artifact_pk)
     filename = CollectionFilename(expected_namespace, expected_name, expected_version)
-    log.info(f"Processing collection from {artifact.file.name}")
+    log.info(f"Processing collection from {temp_file_path}")
     import_logger = logging.getLogger("pulp_ansible.app.tasks.collection.import_collection")
 
-    with _artifact_guard(artifact):
-        try:
-            with artifact.file.open() as artifact_file:
-                importer_result = process_collection(
-                    artifact_file, filename=filename, logger=import_logger
-                )
+    try:
+        with open(temp_file_path, "rb") as artifact_file:
+            importer_result = process_collection(
+                artifact_file, filename=filename, logger=import_logger
+            )
 
-        except ImporterError as exc:
-            log.info(f"Collection processing was not successfull: {exc}")
-            raise
+    except ImporterError as exc:
+        log.info(f"Collection processing was not successfull: {exc}")
+        os.remove(temp_file_path)
```

Comment:
We need to remove the temp file in all cases not only `ImporterError`. To do that I'd suggest replacing `ImporterError` with the more general `Exception`.

Reply:
This comment is really for L#132